persistence/src/androidMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabase.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.persistence.db
+
+import app.cash.sqldelight.driver.android.AndroidSqliteDriver
+import com.wire.kalium.persistence.GlobalDatabase
+import com.wire.kalium.persistence.db.support.SqliteCallback
+import com.wire.kalium.persistence.db.support.SupportOpenHelperFactory
+import com.wire.kalium.persistence.util.FileNameUtil
+import kotlinx.coroutines.CoroutineDispatcher
+
+actual fun globalDatabaseProvider(
+    platformDatabaseData: PlatformDatabaseData,
+    queriesContext: CoroutineDispatcher,
+    passphrase: GlobalDatabaseSecret?,
+    enableWAL: Boolean,
+    encryptionEnabled: Boolean
+): GlobalDatabaseBuilder {
+    val schema = GlobalDatabase.Schema
+    val dbName = FileNameUtil.globalDBName()
+    val driver = if (encryptionEnabled) {
+        System.loadLibrary("sqlcipher")
+        AndroidSqliteDriver(
+            schema = schema,
+            context = platformDatabaseData.context,
+            name = dbName,
+            factory = SupportOpenHelperFactory(passphrase!!.value, enableWAL)
```

Comment:
Can we check if passphrase is not null well as well in the if statement instead of using `!!` operator

Reply:
Is it possible to have `encryptionEnabled` without a password?

Maybe just... 
```kotlin
val isEncryptionEnabled = passphrase != null
```

Not sure why `encryptionEnabled` exists at the moment.

Reply:
I see this can be addressed on the second PR:

- https://github.com/wireapp/kalium/pull/2751/files

Reply:
Good idea! addressed in the second PR 
https://github.com/wireapp/kalium/pull/2751/commits/03321f090cdfd6f644ebdbf16efedd94cf062c1a#diff-a9e521d685cf11ade66a874541e83788285f6ec94c5974512d9e37cb139dbae5